### PR TITLE
peepmatic: add reminder by a power of two rule

### PR DESCRIPTION
**Pull request by MaxGraey**
This PR add optimization rule for peepmatic DSL which simplify reminder with pow of two constants similar to `udiv` rule.
Test cases already exists.

- [ ] This has been discussed in issue #..., or if not, please tell us why
  here.
- [x] A short description of what this does, why it is needed; if the
  description becomes long, the matter should probably be discussed in an issue
  first.
- [x] This PR contains test cases, if meaningful.
- [ ] A reviewer from the core maintainer team has been assigned for this PR.
  If you don't know who could review this, please indicate so. The list of
  suggested reviewers on the right can help you.

### CODE REVIEW

File: cranelift/codegen/src/preopt.peepmatic [hunk 1]
```diff
@@ -191,3 +191,8 @@
 (=> (when (udiv_imm $C $x)
           (is-power-of-two $C))
     (ushr_imm $(log2 $C) $x))
+
+;; Reminder by a power of two -> bitwise and with decreased by one constant.
+(=> (when (urem_imm $C $x)
+          (is-power-of-two $C))
+    (band_imm $(iadd -1 $C) $x))
```
Comment (bjorn3) [inline cranelift/codegen/src/preopt.peepmatic:198]: nit: missing trailing newline

Comment (MaxGraey) [reply cranelift/codegen/src/preopt.peepmatic]: fixed

**Comment by MaxGraey** [comment]
Rewrote to `$(isub 1 $C)` but still got test fails in reason: 
`FAIL filetests/filetests/peepmatic/rem_by_const_power_of_2.clif: panicked in worker #1: not two integers of the same width`

**Comment by MaxGraey** [comment]
It seems macro for folding integers accept only case when both constants have the same bitwidth:
https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/peepmatic/crates/runtime/src/optimizer.rs#L90
Isn't this too restrict?

**Comment by fitzgen** [comment]
> It seems macro for folding integers accept only case when both constants have the same bitwidth:
> https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/peepmatic/crates/runtime/src/optimizer.rs#L90
> Isn't this too much restriction?

This restriction is from cranelift's IR's type system. It won't automatically insert sign- or zero-extension (it wouldn't know which to choose) or truncation. Therefore, we don't need to handle such a scenario as input, and we shouldn't generate it as output.

File: cranelift/codegen/src/preopt.peepmatic [hunk 2]
```diff
@@ -191,3 +191,9 @@
 (=> (when (udiv_imm $C $x)
           (is-power-of-two $C))
     (ushr_imm $(log2 $C) $x))
+
+;; Remainder by a power of two -> bitwise and with decreased by one constant.
+(=> (when (urem_imm $C $x)
+          (is-power-of-two $C)
+          (fits-in-native-word $C))
+    (band_imm $(isub 1 $C) $x))
```
[the remark this answers is not in the document]
Comment (MaxGraey) [reply cranelift/codegen/src/preopt.peepmatic:199]: I see, thanks! Will fix this later

**Pull request merged**